### PR TITLE
adding keyboard: kodachi50

### PR DESCRIPTION
**Pull request by worldspawn00**
## Description

adding keyboard: kodachi50

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wsk/kodachi50/config.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#ifndef CONFIG_H
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/config.h]:
```suggestion
```

File: keyboards/wsk/kodachi50/config.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#define CONFIG_H
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/config.h]:
```suggestion
#pragma once
```

File: keyboards/wsk/kodachi50/config.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#define RGB_DI_PIN D3
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 10
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+#endif
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/config.h]:
```suggestion
```

File: keyboards/wsk/kodachi50/config.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 7
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/config.h]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]: This can all be removed and replaced with the suggestion below.

File: keyboards/wsk/kodachi50/rules.mk [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
MOUSEKEY_ENABLE ?= yes	# Mouse keys
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
EXTRAKEY_ENABLE ?= yes	# Audio control and System control
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
CONSOLE_ENABLE ?= no	# Console for debug
```

File: keyboards/wsk/kodachi50/kodachi50.h [hunk 10]
```diff
@@ -0,0 +1,22 @@
+#ifndef KB_H
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/kodachi50.h]:
```suggestion
```

File: keyboards/wsk/kodachi50/kodachi50.h [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+#define KB_H
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/kodachi50.h]:
```suggestion
#pragma once
```

File: keyboards/wsk/kodachi50/kodachi50.h [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+	{ K10,   K11,   K12,   K13,   K14,   K15,   K16 }, \
+	{ K20,   K21,   K22,   K23,   K24,   K25,   K26 }, \
+	{ K30,   K31,   K32,   KC_NO, K34,   K35,   K36 }, \
+	{ K40,   K41,   K42,   K43,   K44,   KC_NO, K46 }, \
+	{ K50,   K51,   K52,   K53,   K54,   K55,   K56 }, \
+	{ K60,   K61,   K62,   K63,   K64,   K65,   K66 }, \
+	{ K70,   K71,   K72,   K73,   K74,   K75,   K76 }  \
+}
+
+#endif
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/kodachi50.h]:
```suggestion
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
BOOTLOADER = caterina
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
COMMAND_ENABLE = no    # Commands for debug and configuration
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+AUDIO_ENABLE ?= no
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
AUDIO_ENABLE = no
```

File: keyboards/wsk/kodachi50/rules.mk [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+RGBLIGHT_ENABLE ?= yes
```
Comment (ridingqwerty) [inline keyboards/wsk/kodachi50/rules.mk]:
```suggestion
RGBLIGHT_ENABLE = yes
```

File: keyboards/wsk/kodachi50/config.h [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/wsk/kodachi50/config.h]: This block should be removed.  The `IS_COMMAND` is set to the default, and the defaults are set.  

And `PREVENT_STUCK_MODIFIERS` is the default behavior, now.

File: keyboards/wsk/kodachi50/readme.md [hunk 21]
```diff
@@ -0,0 +1,16 @@
+# Kodachi50
+
+![Kodachi50](https://i.imgur.com/v1SYb6a.png)  
+
+The Kodachi50 is a mirror stagger 50% board.  
+
+Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)
```
Comment (drashna) [inline keyboards/wsk/kodachi50/readme.md]:
```suggestion
* Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)  
```

File: keyboards/wsk/kodachi50/readme.md [hunk 22]
```diff
@@ -0,0 +1,16 @@
+# Kodachi50
+
+![Kodachi50](https://i.imgur.com/v1SYb6a.png)  
+
+The Kodachi50 is a mirror stagger 50% board.  
+
+Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)  
+Hardware Supported: The Kodachi50 PCB and kit.
```
Comment (drashna) [inline keyboards/wsk/kodachi50/readme.md]:
```suggestion
* Hardware Supported: The Kodachi50 PCB and kit.  
```

File: keyboards/wsk/kodachi50/readme.md [hunk 23]
```diff
@@ -0,0 +1,16 @@
+# Kodachi50
+
+![Kodachi50](https://i.imgur.com/v1SYb6a.png)  
+
+The Kodachi50 is a mirror stagger 50% board.  
+
+Keyboard Maintainer: [worldspawn00](https://github.com/worldspawn00/)  
+Hardware Supported: The Kodachi50 PCB and kit.  
+Hardware Availability: [PCB & Kit](https://geekhack.org/index.php?topic=100535)
```
Comment (drashna) [inline keyboards/wsk/kodachi50/readme.md]:
```suggestion
* Hardware Availability: [PCB & Kit](https://geekhack.org/index.php?topic=100535)  
```

**Comment by ridingqwerty** [comment]
Committed collaborator suggestions, requesting another look from @drashna

**Comment by ridingqwerty** [comment]
Added a missing keycode to top row of layer 2, Travis is happy now.

**Pull request merged**